File: Strings/1273/1273.c
#include <stdio.h>
#include <string.h>

int max(int a,int b){
    if(a>b)
        return a;
    else
        return b;
}

int main(){
    int N, i,maior_palavra = 0, j,t;
    char palavras[50][50];

    scanf("%d",&N);
    while(N != 0){
        
        maior_palavra = 0;

        for(i = 0; i < N; i++){
            scanf("%s",palavras[i]);
            maior_palavra = max(maior_palavra,strlen(palavras[i]));   
        }

        for(i = 0; i < N; i++){
            t = maior_palavra - strlen(palavras[i]);
           for(j=0;j<t;j++){
                printf(" ");
            }
            printf("%s\n",palavras[i]);
        } 
        scanf("%d",&N);
        if(N != 0)
        printf("\n");
        
    }
    
    return 0;

}

File: Python/Strings/1273.py
# Problema 1273 - Beecrowd - Iniciante - Nível 3

# Quantidade de palavras na primeira iteração
quantidadePalavras = int(input())

# Flag necessário para printar as linhas entre os casos
primeiro = True

while True:
    quantidadePalavras = int(input())

    if quantidadePalavras == 0:
        break

    # Vamos printar a linha antes de cada caso, já que ela não está presente no fim
    # O primeiro caso não tem nada antes
    if primeiro:
        primeiro = False
    else:
        print()

    # Palavras que serão justificadas
    palavras = []

    # Número de caracteres da maior palavra
    maiorPalavra = 0

    # Recebe as palavras na entrada padrão
    for i in range(quantidadePalavras):
        palavra = input()
        
        palavras.append(palavra)

        # Atualiza a quantidade de caracteres da maior palavra
        maiorPalavra = max(maiorPalavra, len(palavra))

    for palavra in palavras:
        # Espaços necessários para justificar a palavra
        print(palavra.rjust(maiorPalavra))
